### PR TITLE
1830 Index local subject headings

### LINKED ISSUE #1830: Verify that local subject terms are being indexed
Are the local terms currently being index.  Subfield 2 local.  If they are not currently being index we need to add them into the index.

we current index (AAT, RBM*, LCGFT, LCSH, Siku)

`650 _7 $2local $5 NjP`

https://www.loc.gov/standards/sourcelist/subject.html

### PR DESCRIPTION
**Pull request by sandbergja**
closes #1830

### CODE REVIEW

File: marc_to_solr/lib/traject_config.rb [hunk 1]
```diff
@@ -830,14 +830,20 @@
 end
 
 to_field 'siku_subject_display' do |record, accumulator|
-  genres = process_hierarchy(record, '650|*7|abcvxyz', ['sk'])
+  genres = process_hierarchy(record, '650|*7|abcvxyz') { |field| siku_heading? field }
```
Comment (mzelesky) [inline marc_to_solr/lib/traject_config.rb:833]: Siku is a subject vocabulary, not a genre vocabulary. It's a minor point, and I know it was called 'genres' in the main branch, but I just wanted to point that out.

File: marc_to_solr/lib/princeton_marc.rb [hunk 2]
```diff
@@ -269,17 +269,16 @@ def set_pub_citation(record)
 
 # for the hierarchical subject/genre display
 # split with em dash along t,v,x,y,z
-# optional vocabulary argument for allowing certain subfield $2 vocabularies
-def process_hierarchy(record, fields, vocabulary = [])
+# optionally pass a block to only allow fields that match certain criteria
```
Comment (mzelesky) [inline marc_to_solr/lib/princeton_marc.rb:272]: Could you add a block here in the comments to provide an example?

**Review by mzelesky — CHANGES_REQUESTED** [review]
@sandbergja I reviewed this with @christinach , and there were a few minor questions. Thank you, everything else looks great!

**Review by mzelesky — APPROVED** [review]
Thank you for addressing the minor points I had. Looks good to me.

**Review by christinach — APPROVED** [review]

**Pull request merged**